ffcx/codegeneration/numba/implementation.py

```diff
@@ -137,21 +157,23 @@ def format_binary_op(self, oper: L.BinOp) -> str:
         # Return combined string
         return f"{lhs} {oper.op} {rhs}"
 
-    def format_neg(self, val: L.Neg) -> str:
+    @_format.register
+    def _(self, val: L.Neg) -> str:
         """Format unary negation."""
-        arg = self.format(val.arg)
+        arg = self(val.arg)
         return f"-{arg}"
 
-    def format_not(self, val: L.Not) -> str:
+    @_format.register
+    def _(self, val: L.Not) -> str:
         """Format not operation."""
-        arg = self.format(val.arg)
+        arg = self(val.arg)
         return f"not({arg})"
 
-    def format_andor(self, oper: L.And | L.Or) -> str:
+    def _format_and_or(self, oper: L.And | L.Or) -> str:
```

Comment:
Same comment as for the C implementation regarding using multiple dispatch on unions.